tests/unittests/common/lengthtest.cpp

```diff
@@ -68,6 +68,17 @@ TEST_P(LengthTest, testToMmString) {
   }
 }
 
+TEST(LengthTest, testSnapToGrid) {
+  EXPECT_EQ(Length(8).mappedToGrid(10), Length(10));
+  EXPECT_EQ(Length(2).mappedToGrid(10), Length(0));
+  EXPECT_EQ(Length(-8).mappedToGrid(10), Length(-10));
+  EXPECT_EQ(Length(-2).mappedToGrid(10), Length(0));
+  EXPECT_EQ(Length(18).mappedToGrid(10), Length(20));
+  EXPECT_EQ(Length(12).mappedToGrid(10), Length(10));
+  EXPECT_EQ(Length(-18).mappedToGrid(10), Length(-20));
+  EXPECT_EQ(Length(-12).mappedToGrid(10), Length(-10));
```

Comment:
Could you maybe add some tests for following cases?
- Input Length == 0
- Grid Interval == 0
- Grid Interval == 1
- Input Length == Grid Interval
- Input Length == Multiple of Grid Interval

Reply:
Done, it looks like it didn't work well with Grid Interval == 1 :smile: